### PR TITLE
Use single centralized version catalog for all samples.

### PR DESCRIPTION
**Pull request by JuliaJakubcova**

### CODE REVIEW

File: samples/sample-bugsnag/settings.gradle.kts [hunk 1]
```diff
@@ -12,6 +12,14 @@ include(":app", ":shared")
 
 enableFeaturePreview("VERSION_CATALOGS")
```
Comment (psh) [inline samples/sample-bugsnag/settings.gradle.kts:13]: If you update the version of gradle, you can drop this feature preview in all the files as _version catalogs_ are now a stable gradle feature.

**Comment by samhill303** [comment]
There is a `readParentKotlin` function used in a few places to "centralize" that version. Can that be removed somehow in favor of what you have here?

**Comment by samhill303** [comment]
There are also a handful of versions defined in `gradle.properties` (stately,crashkios,etc) it would be good to move. Also `classpath("com.android.tools.build:gradle:7.2.2")` in places

**Comment by mrf7** [comment]
I don't care about it enough to actually requests changes/block merge but I'm partial the syntax of defining the dependencies in settings.gradle rather than in toml. Just my two cents so can be something to discuss later and maybe iterate

**Comment by russhwolf** [comment]
toml has bad tooling support. settings.gradle has bad encapsulation. You can't win either way.

**Review by russhwolf — APPROVED** [review]
Check the crashkios versions, but my other comments are just style nitpicks so don't worry if you don't want to change anything else.

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+firebase-crashlytics-gradle = "2.7.1"
+
+bugsnag-android-gradle-plugin = "7.0.0"
+
+stately = "1.2.3"
+testhelp = "0.6.4"
+
+crashkios = "0.8.1"
+crashkios-bugsnag = "0.7.1-alpha3"
+crashkios-utils = "0.7.1-alpha3"
```
[the remark this answers is not in the document]
Comment (JuliaJakubcova) [reply gradle/libs.versions.toml:31]: Newest version of Crashkios is 0.8.1, but it seems it was not published for crashkios utils and bugsnag :/

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+koin-android = { module = "io.insert-koin:koin-android", version.ref = "koin" }
+coroutines = { module = "org.jetbrains.kotlinx:kotlinx-coroutines-core", version.ref = "coroutines" }
+roboelectric = { module = "org.robolectric:robolectric", version.ref = "roboelectric" }
+
+koin-test = { module = "io.insert-koin:koin-test", version.ref = "koin" }
+coroutines-test = { module = "org.jetbrains.kotlinx:kotlinx-coroutines-test", version.ref = "coroutines" }
+
+android-junitTest = { module = "androidx.test.ext:junit", version.ref = "android-junitTest" }
+junitTest = { module = "junit:junit", version.ref = "junit" }
+maven-publish = { module = "com.vanniktech:gradle-maven-publish-plugin", version.ref = "mavenPublish" }
```
[the remark this answers is not in the document]
Comment (JuliaJakubcova) [reply gradle/libs.versions.toml:82]: When defining plugins they cannot (or I didnt find a way) contain ":", so it cannot be used there.

File: gradle/libs.versions.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (samhill303) [reply gradle/libs.versions.toml:31]: This matches what we had before this PR. So made a new ticket to follow up on getting the latest versions published and added https://github.com/touchlab/Kermit/issues/318

**Pull request merged**